### PR TITLE
[stdlib] YAIAPR

### PR DESCRIPTION
**Pull request by airspeedswift**
* Remove `Comparable` conformance for `_SwiftNSOperatingSystemVersion`
* Lazy performance needs specialization/inlining
* Bool inlining consistency
* Uninline dump/stdout
* Uninline printing functions
* Clean up very outdated CollectionOld test

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark staging

File: stdlib/public/core/OutputStream.swift [hunk 1]
```diff
@@ -415,7 +414,7 @@ internal func _print_unlocked<T, TargetStream : TextOutputStream>(
 ///
 /// This function is forbidden from being inlined because when building the
 /// standard library inlining makes us drop the special semantics.
-@inline(never) @_effects(readonly)
```
Comment (jrose-apple) [inline stdlib/public/core/OutputStream.swift]: Is this no longer relevant?

Comment (airspeedswift) [reply stdlib/public/core/OutputStream.swift]: Yeah. I'm going to do a pass where I remove all `@inline(never)` calls that aren't also explicitly `@inlinable` in the next one.

**Pull request merged**